xonsh/shell.py

```diff
@@ -11,6 +11,20 @@
 from xonsh.platform import (best_shell_type, has_prompt_toolkit,
                             ptk_version_is_supported)
 from xonsh.tools import XonshError, to_bool_or_int
+from xonsh.events import events
+
+
+events.doc('on_precommand', """
```

Comment:
I am in favor of changing these to just `{pre,post}command` since the `on_` is implied. It is your PR so I defer to your naming, but I thouhgt i would register my preference.